tom_targets/sharing.py

```
@@ -4,7 +4,66 @@
 from django.core.exceptions import ImproperlyConfigured
 
 from tom_targets.serializers import TargetSerializer
-from tom_dataproducts.sharing import get_destination_target
+from tom_targets.models import PersistentShare
+from tom_dataproducts.sharing import check_for_share_safe_datums, share_data_with_tom, get_destination_target
+from tom_dataproducts.models import ReducedDatum
+from tom_dataproducts.alertstreams.hermes import publish_to_hermes, BuildHermesMessage
+
+
+def share_target_and_all_data(share_destination, target):
```

This is silently breaking for me.
I'm getting back `b'{"message_text":["This field may not be null."],"submitter":["This field may not be blank."]}'`
Obviously these things need to be autofilled or use the other form.
It would also be nice if these made the same user facing alerts that normal sharing does (for success as well as failure.)

---

Thanks for checking that - I had thought submitter would be autoset by Hermes but it actually doesn't do that for outgoing messages, only incoming. So I've set the submitter to be the `TOM_NAME` which seems appropriate to me, and set the message text to be empty string instead of None.

---

Also, the error should me mapped through now so it should stop the creation of a persistent share and show the error on screen if you have a problem when sharing all existing data. If it succeeds, nothing is shown but the persistent share is created and will show up, so that is the signal that it succeeded in sharing existing data too.

---

I'm still getting a silent failure for `HTTPError('400 Client Error: Bad Request for url: https://hermes-dev.lco.global/api/v0/submit_message/')
b'{"topic":["Hermes Dev can only submit to the hermes.test topic."]}'`

as well as missing telescope/instrument

---

Hmm I'll look into the error not making it through... What do you mean by missing telescope/instrument though? Is it stored in your datums value dict as 'telescope' and 'instrument'?

---

yeah, not having the telescope/instrument in my data is just another way my hermes message can be rejected. The issue isn't that, it's just one of the two ways I tried to get errors back from hermes.

---

Ah okay I understand what were saying now

---

I've updated it again, and the errors are definitely coming through now! Sorry the error feedback handler stuff had me confused a bit.